src/selectors/entities/channels.ts

```diff
@@ -795,6 +795,80 @@ export const getSortedUnreadChannelIds: (state: GlobalState, lastUnreadChannel:
     (unreadChannelIds, mappedAndSortedUnreadChannelIds) => mappedAndSortedUnreadChannelIds,
 );
 
+//recent channels
+
+export const getRecentChannelIds: (state: GlobalState) => Array<string> = createIdsSelector(
+    getAllChannels,
+    getMyChannelMemberships,
+    getChannelIdsForCurrentTeam,
+    (channels: IDMappedObjects<Channel>, members: RelationOneToOne<Channel, ChannelMembership>, teamChannelIds: Array<string>): Array<string> => {
+        const recentIds = teamChannelIds.filter((id) => {
+            const c = channels[id];
+            const m = members[id];
+
+            if (c && m) {
+                return true;
+            }
+
+            return false;
+        });
+
+        return recentIds;
+    },
+);
+
+export const getRecentChannels: (state: GlobalState) => Array<Channel> = createIdsSelector(
+    getCurrentUser,
+    getUsers,
+    getUserIdsInChannels,
+    getAllChannels,
+    getRecentChannelIds,
+    getTeammateNameDisplaySetting,
+    (currentUser, profiles, userIdsInChannels: any, channels, recentIds, settings) => {
+        // If we receive an unread for a channel and then a mention the channel
+        // won't be sorted correctly until we receive a message in another channel
+        if (!currentUser) {
+            return [];
+        }
+
+        const allRecentChannels = recentIds.filter((id) => channels[id] && channels[id].delete_at === 0).map((id) => {
+            const c = channels[id];
+
+            if (c.type === General.DM_CHANNEL || c.type === General.GM_CHANNEL) {
+                return completeDirectChannelDisplayName(currentUser.id, profiles, userIdsInChannels[id], settings!, c);
+            }
+
+            return c;
+        });
+        return allRecentChannels;
+    },
+);
+
+export const mapAndSortRecentChannelIds = (
+    channels: Array<Channel>,
+    currentUser: UserProfile,
+    lastPosts: RelationOneToOne<Channel, Post>,
+    sorting: SortingType,
+): Array<string> => {
+    const locale = currentUser.locale || General.DEFAULT_LOCALE;
+
+    const recentChannelIds = channels.
+        sort(sortChannelsByRecencyOrAlpha.bind(null, locale, lastPosts, sorting)).
+        map((channel) => channel.id);
+
+    return recentChannelIds;
+};
+
+export const getAllRecentChannelIds: (state: GlobalState, sorting: SortingType) => Array<string> = createIdsSelector(
```

Comment:
It might be simpler to avoid some of the intermediate functions here if that makes your life easier. Part of the reason that we have multiple methods for `getUnreadChannels`, `getUnreadChannelIds`, `getUnreadChannelsSet`, etc for other types of channels is a bit of a legacy issue.

I'm not sure which version of the selector would be needed for the channel switcher modal in the web app. Do you have a webapp PR that you're working on that'll use these changes?

Reply:
Yes I have combined all the different functions into a single selector  and I need this selector to get data the recent channels to display in the switcher for this issue https://github.com/mattermost/mattermost-server/issues/14633, I have to raise a PR for the web app also but once this redux selector is ready I will raise a PR for the web app.

![image](https://user-images.githubusercontent.com/44333182/88766276-5e42f480-d195-11ea-943d-6090c977ee2e.png)